subiquity/server/ubuntu_advantage.py

```diff
@@ -132,20 +133,31 @@ async def query_info(self, token: str) -> dict:
             "--format", "json",
             "--simulate-with-token", token,
         )
-        try:
-            proc: CompletedProcess = await utils.arun_command(command,
-                                                              check=True)
+
+        # On error, the command will exit with status 1. When that happens, the
+        # output should still be formatted as a JSON object and we can inspect
+        # it to know the reason of the failure. This is how we figure out if
+        # the contract token was invalid.
+        proc: CompletedProcess = await utils.arun_command(command, check=False)
+        if proc.returncode == 0:
             # TODO check if we're not returning a string or a list
-            return json.loads(proc.stdout)
-        except CalledProcessError:
+            try:
+                return json.loads(proc.stdout)
+            except json.JSONDecodeError:
+                log.exception("Failed to parse output of command %r", command)
+        elif proc.returncode == 1:
+            try:
+                data = json.loads(proc.stdout)
+            except json.JSONDecodeError:
+                log.exception("Failed to parse output of command %r", command)
+            else:
+                with contextlib.suppress(KeyError):
+                    for error in data["errors"]:
+                        if error["message_code"] == "attach-invalid-token":
```

Comment:
Is there anything else interesting here?  Would be nice to log.debug() it.

Reply:
```
2022-07-06 14:40:57,301 DEBUG subiquitycore.utils:101 arun_command ('/usr/bin/ubuntu-advantage', 'status', '--format', 'json', '--simulate-with-token', 'C32454555555555555555555') exited with code 1
2022-07-06 14:40:57,301 DEBUG subiquity.server.ubuntu_advantage:159 error reported by u-a-c: attach-invalid-token: Invalid token. See https://ubuntu.com/advantage
```

Reply:
Thanks!